src/main/java/at/hannibal2/skyhanni/features/garden/farming/LaneswitchNotification.kt

```diff
@@ -0,0 +1,76 @@
+package at.hannibal2.skyhanni.features.garden.farming
+
+import at.hannibal2.skyhanni.config.features.garden.laneswitch.LaneswitchFarmConfig
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.utils.LocationUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.sendTitle
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SoundUtils
+import at.hannibal2.skyhanni.utils.SoundUtils.playSound
+import kotlin.time.Duration.Companion.seconds
+
+class LaneswitchNotification {
+
+    private val config get() = GardenAPI.config.laneswitch
+
+    private var blocksPerSecond = 0.0
+    private var lastPos = LorenzVec(0, 0, 0)
+    private var lastLaneSwitch = SimpleTimeMark.farPast()
+
+    private fun coverable(from: LorenzVec, to: LorenzVec, speed: Double, time: Int): Boolean {
+        return from.distance(to) <= speed * time
+    }
+
+    companion object {
+        private val config get() = GardenAPI.config.laneswitch
+
+        @JvmStatic
+        fun playUserSound() {
+            SoundUtils.createSound(
+                config.notification.sound.notificationSound,
+                config.notification.sound.notificationPitch,
+                config.notification.sound.notificationVolume
+            ).playSound()
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!config.enabled || !GardenAPI.inGarden()) return
+        if (GardenCropSpeed.averageBlocksPerSecond <= 0.0 && config.notification.farmingOnly) return
+
+        val playerPosition = LocationUtils.playerLocation().round(2)
+        blocksPerSecond = playerPosition.distance(lastPos) * 20
+        this.lastPos = playerPosition
+
+        if (lastLaneSwitch.passedSince() >= config.notification.settings.notificationTimeout.seconds) {
+            val farmEnd: List<LorenzVec> = when (config.farm.farmDirection) {
+                LaneswitchFarmConfig.FarmDirection.NORTH, LaneswitchFarmConfig.FarmDirection.SOUTH -> {
+                    val xValue = (config.farm.plotAmount * 48).toDouble()
```

Comment:
This way of finding plot boundaries feels like it breaks very easily for `plot length != 5`. Maybe consider using the existing `GardenAPI` to find unlocked plots in the current row/column.